File: knowledge_base_service/tests/fixtures/code_samples/sample.ts
/**
 * Person interface
 */
interface IPerson {
    name: string;
    age: number;
    greet(): string;
}

/**
 * Person class implementation
 */
class Person implements IPerson {
    constructor(
        public name: string,
        public age: number
    ) {}

    greet(): string {
        return `Hello, my name is ${this.name}`;
    }

    celebrateBirthday(): void {
        this.age++;
    }

    getName(): string {
        return this.name;
    }
}

/**
 * Calculator with generic type
 */
class Calculator<T extends number> {
    private history: T[] = [];

    add(a: T, b: T): T {
        const result = (a + b) as T;
        this.history.push(result);
        return result;
    }

    subtract(a: T, b: T): T {
        const result = (a - b) as T;
        this.history.push(result);
        return result;
    }

    getHistory(): T[] {
        return [...this.history];
    }
}

// Type alias
type Processor<T> = (input: T) => T;

// Standalone function
function processData<T>(data: T[]): T[] {
    return data.filter(item => item !== null);
}

// Arrow function with type annotation
const multiply: (a: number, b: number) => number = (a, b) => a * b;

export { Person, Calculator, processData, multiply };

File: knowledge_base_service/tests/fixtures/code_samples/Sample.java
package com.example.test;

import java.util.List;
import java.util.ArrayList;
import java.util.Optional;

/**
 * A person class for testing.
 */
public class Person {
    private String name;
    private int age;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String greet() {
        return "Hello, my name is " + name;
    }

    public void celebrateBirthday() {
        age++;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }
}

/**
 * Calculator class with basic operations.
 */
class Calculator {
    private List<Double> history = new ArrayList<>();

    public double add(double a, double b) {
        double result = a + b;
        history.add(result);
        return result;
    }

    public double subtract(double a, double b) {
        double result = a - b;
        history.add(result);
        return result;
    }

    public List<Double> getHistory() {
        return new ArrayList<>(history);
    }
}

/**
 * Utility interface.
 */
interface Processor<T> {
    T process(T input);
}
